File: payment-intent/python/payment-intent-auto.py
import http.client
import json

conn = http.client.HTTPSConnection("api.dojo.tech")
payload = json.dumps({
  "amount": {
    "value": 1000,
    "currencyCode": "GBP"
  },
  "reference": "Order 234"
})
headers = {
  'version': '2022-07-04',
  'Content-Type': 'application/json',
  'Accept': 'application/json',
  'Authorization': 'Basic sk_sandbox_c8oLGaI__msxsXbpBDpdtwJEz_eIhfQoKHmedqgZPCdBx59zpKZLSk8OPLT0cZolbeuYJSBvzDVVsYvtpo5RkQ'
}
conn.request("POST", "/payment-intents/", payload, headers)
res = conn.getresponse()
data = res.read()
print(data.decode("utf-8"))
conn.close()

File: payment-intent/cs/payment-intent-auto.cs
using RestSharp;
var client = new RestClient("https://api.dojo.tech/payment-intents/");
client.Timeout = -1;
var request = new RestRequest(Method.POST);
request.AddHeader("version", "2022-05-26");
request.AddHeader("Content-Type", "application/json");
request.AddHeader("Accept", "application/json");
request.AddHeader("Authorization", "Basic sk_sandbox_c8oLGaI__msxsXbpBDpdtwJEz_eIhfQoKHmedqgZPCdBx59zpKZLSk8OPLT0cZolbeuYJSBvzDVVsYvtpo5RkQ");
var body = @"{" + "\n" +
@"    ""amount"": {" + "\n" +
@"        ""value"": 1000," + "\n" +
@"        ""currencyCode"": ""GBP""" + "\n" +
@"    }," + "\n" +
@"    ""tipsAmount"": {" + "\n" +
@"        ""value"": 10," + "\n" +
@"        ""currencyCode"": ""GBP""" + "\n" +
@"    }," + "\n" +
@"    ""reference"": ""Order 234""," + "\n" +
@"    ""description"": ""Demo payment intent""," + "\n" +
@"    ""captureMode"": ""Auto""," + "\n" +
@"    ""details"": {" + "\n" +
@"        ""showTotal"": ""true""," + "\n" +
@"        ""showReference"": ""true""" + "\n" +
@"    }," + "\n" +
@"    ""config"": {" + "\n" +
@"        ""payment"": {" + "\n" +
@"            ""customAmountAllowed"": ""true""," + "\n" +
@"            ""tipsAllowed"": ""true""" + "\n" +
@"        }" + "\n" +
@"    }" + "\n" +
@"}";
request.AddParameter("application/json", body,  ParameterType.RequestBody);
IRestResponse response = client.Execute(request);
Console.WriteLine(response.Content);